### PR TITLE
Thunderscope: HRVO Simulator Visualization Layer

### LINKED ISSUE #2499: Thunderscope: Add HRVO visualization field overlay
### Description of the task
We would like to be able visualize the HRVO simulator in Thunderscope. This includes visualizing the velocity obstacles with two lines or a triangle (polygon) from the perspective of any friendly robot, the Agent positions and radii. Currently `HRVOSimulator::visualize()` exists which uses an `TbotsProto::Obstacle` to visualize the HRVO Simulator (although it is commented out), however, this protobuf is already being used by the navigator, which results in Thunderscope flashing between the obstacles included in the HRVO and Navigator `LOG(VISUALIZE)` calls.   

### Acceptance criteria
Here is one approach. Feel free to find better/easier implementations.
- [ ] Create a new protobuf `HRVOSimulator` message which includes a list of velocity obstacles from the perspective of one Agent (represented as a list of line pairs or a list of polygons, look at `Agent::VelocityObstacle` to see how a velocity obstacle is currently presented) and friendly/enemy Agents (represented as Circles). Note that each Agent has a different set of VelocityObstacles. So to make the visualization easier to comprehend with less overlaying velocity obstacles, we would want to only look at the set of VelocityObstacles of one HRVO Agent.  
- [ ] Create a new Protobuf message which includes a map of friendly robot ids to the robot's `HRVOSimulator` protobuf message (explained above).
- [ ] Create an overlay to the field widget which has a selection to choose between robot ids to showcase one (optionally can show more) of the HRVO simulator protobuf messages. The implementation of this should be very similar to the implementation of the existing Obstacle overlay.

### PR DESCRIPTION
**Pull request by nimazareian**
### Description
Added HRVOLayer to thunderscope to visualize the current state of the HRVO simulator (i.e. Agent position + radius represented with circles, and velocity obstacles seen by the robot which owns that HRVO simulator). Since each friendly robot has there own HRVO Simulator, there are multiple layers added (one per robot). 
Note that the layers are hidden by default (to avoid filling of the field), and can be enabled through the legend.

### Testing Done
Manually testing cpp and py tests (+ with replay playback)

### Resolved Issues
resolves #2499 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/simulated_tests/simulated_er_force_sim_play_test_fixture.cpp [hunk 1]
```diff
@@ -85,6 +85,7 @@ void SimulatedErForceSimPlayTestFixture::updatePrimitives(
 
     auto primitive_set_msg = ai.getPrimitives(world_with_updated_game_state);
     LOG(VISUALIZE) << ai.getPlayInfo();
+    LOG(VISUALIZE) << *primitive_set_msg;
```
Comment (jonathanlew) [inline src/software/simulated_tests/simulated_er_force_sim_play_test_fixture.cpp:88]: hm, what is this for?

**Review by jonathanlew — COMMENTED** [review]
Looks good. Why am I still seeing `WARNING [simulator.cpp->visualize:381]	HRVO friendly agent with robot id 1 can not be visualized.` warnings?

File: src/software/simulated_tests/simulated_er_force_sim_play_test_fixture.cpp (hunk 1 above)
Comment (nimazareian) [reply src/software/simulated_tests/simulated_er_force_sim_play_test_fixture.cpp:88]: The PrimitiveSet (path + static obstacles) were never being visualized during C++ tests, so had to add this.

**Comment by nimazareian** [comment]
> Looks good. Why am I still seeing WARNING [simulator.cpp->visualize:381] HRVO friendly agent with robot id 1 can not be visualized. warnings?

@jonathanlew I think in the case of the simulated tests, it happens because we call `PrimitiveExecutor::stepPrimitive` for every robot, at every tick which attempts to visualize that robot, however, robots which are not included in the PrimitiveSet (i.e. not doing anything like robot 0 in `simulated_hrvo_test`s) never get a `PrimitiveExecutor::updateWorld` call and as a result their corresponding HRVO simulator never gets populated with any Agents. As such, the Agent can not be found and visualized.
Though, I guess I'm not sure how useful this log is anyways, so it might be better if we remove it, or reduce its verbosity to `INFO`. Hopefully with #2655, the legend gets updated dynamically and that would be enough to tell the user that the Robot is not being visualized.

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by itsarune — APPROVED** [review]
works locally on my computer (albeit a bit laggy so I upped the buffer size on my end) 👍

**Pull request merged**